grapher/core/LegacyToOwidTable.ts

```diff
@@ -202,16 +212,134 @@ export const legacyToOwidTableAndDimensions = (
+    // then by year+entity and finally entity only.
+    // This last join by entity only is important so that variables that don't have values for years
+    // that come up in the day+entity index we still retain the values. E.g. our continents table
+    // has values for all countries but only for the year 2015. If we join that with covid era days
+    // we still want to retain the continents so we have the fallback to entity only (this was also
+    // the only behaviour prior to July 2022). Remember that tolerance will only be applied much later -
+    // here we are only concerned with merging mutliple variables into an inputTable that retains information.
+    // Another approach would be to convert years into days when we have days - then we could simplify the fallback
+    // join key logic described above.
+    // Another caveat is that by switching to day+entity as the primary index that we use to join we can drop some entities.
+    // This happens e.g. with Antarctica if the continents table is used. Continents contains an entry for the entity Antarctica for 2015
+    // that maps it (and 3 other territories) to the Antarctica continent. If the days variables don't have values for any of the
+    // Antarctica entities then they will not be enumerated for the final join table and thus they will be dropped from the final table.
+    // This is maybe counter to what you would expect from a full join but is simply an artifact of making days+entities the primary
+    // index and not backporting years to days. We might want to revisit this in the future and/or also apply tolerance already
+    // at this level here.
+
+    // Merge all day based variables together (returns an empty table if there are none)
+    const variablesJoinedByDay = fullJoinTables(variableTablesToJoinByDay, [
+        OwidTableSlugs.day,
+        OwidTableSlugs.entityId,
+    ])
+
+    let joinedVariablesTable: OwidTable
+    // If we have both day and year based variables we need to do some special logic as described above
+    if (
+        variableTablesToJoinByYear.length > 0 &&
+        variableTablesToJoinByDay.length > 0
+    ) {
+        // Derive the year from the day column and add it to the joined days table
+        const daysColumn = variablesJoinedByDay.getColumns([
+            OwidTableSlugs.day,
+        ])[0]
+        const yearsForDaysValues = []
+        for (let i = 0; i < daysColumn.values.length; i++) {
+            yearsForDaysValues.push(
+                getYearFromISOStringAndDayOffset(
+                    EPOCH_DATE,
+                    daysColumn.values[i] as number
+                )
+            )
+        }
+        const newYearColumn = {
+            ...daysColumn,
+            slug: OwidTableSlugs.year,
+            name: OwidTableSlugs.year,
+            values: yearsForDaysValues,
+        } as unknown as OwidColumnDef
```

Comment:
I think this should rather be:
```suggestion
        const newYearColumn = {
            ...daysColumn.def,
            slug: OwidTableSlugs.year,
            name: OwidTableSlugs.year,
            values: yearsForDaysValues,
        } as OwidColumnDef
```